src/main/java/org/mvel2/math/MathProcessor.java

```diff
@@ -376,7 +376,9 @@ private static Object doOperationsSameType(int type1, Object val1, int operation
           case DIV:
             return toDouble(val1) / toInteger(val2);
           case MULT:
-            return toInteger(val1) * toInteger(val2);
+            long l = (long) (toDouble(val1) * toInteger(val2));
```

Comment:
This code is very frequently used and is far from perfect. Please optimize it. Do not convert values twice. Do not convert values to double. You may convert integer to long and this will be sufficient. Please improve.